docs/src/proposals/standalone-tpu-node.md

```diff
@@ -0,0 +1,188 @@
+|   | Fetch     |   |       |                      | Fetch     |   |
+|   +-----+-----+   |       |                      +-----+-----+   |
+|         |         |       |                            |         |
+|         |         |       |                            |         |
+|   +-----v-----+   |       |   +------------+     +-----v-----+   |
+|   | SigVerify |   |   +-------> ProxyFetch |     | SigVerify |   |
+|   +-----+-----+   |   |   |   +-----+------+     +-----+-----+   |
+|         |         |   |   |         |                  |         |
+|         |         |   |   |         |                  |         |
+|   +-----v-----+   |   |   |         |            +-----v-----+   |
+|   | Proxy     +-------+   |         +------------> Banking   |   |
+|   +-----------+   |       |                      +-----+-----+   |
+|                   |       |                            |         |
++-------------------+       |                            |         |
+                            |                            v         |
+                            |                           ...        |
+                            |                                      |
+                            +--------------------------------------+
+```
+
+The _TPUProxy_ protocol is based on gRPC. Mutual TLS authentication is _mandatory_.
+
+We propose the following API definition.
+
+```proto
+import "google/protobuf/duration.proto";
+
+// Validator TPUProxy API.
+service TPUProxy {
+    // Use this method to advertise a new TPU endpoint in gossip.
+    // Clients should repeatedly invoke this endpoint before their TPU registration expires
+    // or when a transaction stream terminates.
+    rpc register_tpu_endpoint(RegisterTPURequest) returns (RegisterTPUResponse);
+
+    // Use this method to push a stream of transactions to the validator.
+    rpc stream_transactions(stream Transaction) returns (StreamTransactionsResponse);
+}
+
+message StreamTransactionsResponse {}
+
+message RegisterTPURequest {
+    string ip_address = 1;
+    uint32 port = 2;
+}
+
+message RegisterTPUResponse {
+    // The duration for which the TPU registration is valid.
+    google.protobuf.Duration registration_ttl = 1;
+}
+```
+
+### TPU failover
+
+When the _ProxyFetch_ stage is actively processing transactions from TPU nodes,
+the validator _Fetch_ will cease processing TPU traffic and remove its registration from gossip.
+This effectively re-routes all TPU traffic away from the validator.
+
+If all external TPU recipients fail however, the validator system would stop processing transactions entirely.
+The system should naturally recover in this scenario by reactivating TPU processing in the validator _Fetch_ stage.
+
+This failover is triggered when either of the following conditions are met for 10 seconds.
+- No active TPU registrations (via `register_tpu_endpoint`).
+- No transactions received in the last 30 seconds.
```

Comment:
we were thinking of a heartbeat like using TTL and maybe shorter timeout? 30s will screw with things if you're not a leader for >30s

Reply:
Yeah, I've missed the fact that TPU traffic only comes in during leader slots. Let's remove that condition entirely